src/wp-includes/update.php

```diff
@@ -956,6 +956,51 @@ function wp_clean_update_cache() {
 	delete_site_transient( 'update_core' );
 }
 
+/**
+ * Deletes all contents of the temp-backup directory.
+ *
+ * @since 6.0.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ */
+function wp_delete_all_temp_backups() {
+	/*
+	 * Check if there's a lock, or if currently performing an Ajax request,
+	 * in which case there's a chance we're doing an update.
+	 * Reschedule for an hour from now and exit early.
+	 */
+	if ( get_option( 'core_updater.lock' ) || get_option( 'auto_updater.lock' ) || wp_doing_ajax() ) {
+		wp_schedule_single_event( time() + HOUR_IN_SECONDS, 'wp_delete_temp_updater_backups' );
+		return;
+	}
+
```

Comment:
A logic check to ensure this isn't constantly happening would be good. If it's happened 24 times (one day) it's probably a safe bet it will happen a 25th and that we are invalidating the options cache without reason.

An upper limit will also reduce the chance that at some point both the recurring and single event will run.

Reply:
FWIW I only see a single `wp_delete_temp_updater_backups` cron event.

Reply:
This cron event is already scheduled weekly in `WP_Upgrader::init()`. There's a check for a current cron event there.

Reply:
Logic looks good. This only fires on the weekly cron event and the reschedule is only if there's a current update happening.